File: Demo/Objective-C_Storyboard/LGSideMenuControllerDemo/ChooseNavigationController.m
//
//  ChooseNavigationController.m
//  LGSideMenuControllerDemo
//

#import "ChooseNavigationController.h"

@implementation ChooseNavigationController

- (BOOL)shouldAutorotate {
    return YES;
}

- (BOOL)prefersStatusBarHidden {
    return UIInterfaceOrientationIsLandscape(UIApplication.sharedApplication.statusBarOrientation) && UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPhone;
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (UIStatusBarAnimation)preferredStatusBarUpdateAnimation {
    return UIStatusBarAnimationNone;
}

@end

File: Demo Swift/LGSideMenuControllerDemo/ChooseNavigationController.swift
//
//  ChooseNavigationController.swift
//  LGSideMenuControllerDemo
//
//  Created by Cole Dunsby on 2016-07-25.
//  Copyright © 2016 Cole Dunsby. All rights reserved.
//

import UIKit

class ChooseNavigationController: UINavigationController {

    override func shouldAutorotate() -> Bool {
        return true
    }
    
    override func prefersStatusBarHidden() -> Bool {
        return UI_USER_INTERFACE_IDIOM() == .Phone && UIInterfaceOrientationIsLandscape(UIApplication.sharedApplication().statusBarOrientation)
    }
    
    override func preferredStatusBarStyle() -> UIStatusBarStyle {
        return .LightContent
    }
    
    override func preferredStatusBarUpdateAnimation() -> UIStatusBarAnimation {
        return .None
    }

}
